### ISSUE TITLE
Export functions as methods on the List class

### ISSUE THREAD

**Issue by danr** [funkia/list#22]
Hi paldepind, thanks for making this interesting library.

I wonder why you chose to make functions as stand-alone and not property methods on the List class? This would make them a little more convenient to use, i.e. `xs.filter(p)` instead of `L.filter(xs, p)`.

/Dan

**Comment by paldepind** [comment]
Hello @danr. You're very welcome. I'm happy to hear that you find it interesting 😄 

There are two reasons for the absence of methods:

1. Some people prefer to work with plain functions
2. Methods don't work with tree-shaking. As soon as a class is imported _all_ it's methods are imported as well.

But, I'm planning to optionally offer methods in the next release. The idea is that you can do the following to get methods:

```js
import * as L from "list/methods";
// Now this works
const myList = L.list(0, 1, 2, 3, 4, 5, 6);
cosnt myOtherList = myList.map(f).filter(p).take(3);
```

In other words, when you import `list/methods` you get a variant of List in which every function is available as a method. This gives the best of both worlds. People who don't want methods don't get them and people who really like methods do get them.

I'd like to hear what you think about the idea?

**Comment by danr** [comment]
Ah, I hadn't thought of tree shaking.

Is it possible to do the `list/method` approach without having two different kinds of incompatible `List` around? I.e. you get the normal `List` class but importing `list/method` adds more methods to `List.property`? 

If this can be done without incompatibilities it sounds pretty ideal.

**Comment by paldepind** [comment]
> Is it possible to do the list/method approach without having two different kinds of incompatible List around? I.e. you get the normal List class but importing list/method adds more methods to List.property?

Yes. That's exactly how it works. You can see a tiny [proof of concept here](https://github.com/funkia/list/blob/master/src/methods.ts). That is actually secretly a part of the current release. But it only gives you `append` and `nth` so it's not very useful 😉

**Comment by danr** [comment]
That looks great! :+1:

**Comment by paldepind** [comment]
`list/methods` is now available in the latest release 😄

**Comment by danr** [comment]
Cool! :tada:
